pkg/infra/baseagent.go

```diff
@@ -49,6 +49,14 @@ func (a *Base) HP() int {
 	return a.hp
 }
 
+func (a *Base) CurrPlatFood() float64 {
+	foodOnPlatform, platformFloor := a.tower.GetCurrPlatFood()
+	if platformFloor == a.floor { // only show the food on the platform if the platform is on the same floor as the agent
+		return foodOnPlatform // if agent and platform on same floor, return the amount of food on the platform
+	}
+	return 0.0 // otherwise return nothing because you should not be able to see food on the platform if its not on your floor.
```

Comment:
```suggestion
	return -1.0 // otherwise return -1 to indicate the platform is not on the floor. Agents cannot see the food on the platform if it is not on their floor
```

Reply:
If we return 0 for when the platform is not on the floor, then the agent cannot know if 0 means no food or no platform

Reply:
We can still return 0 and have a separate Boolean function to query if the platform is on the floor or not